lib/Test/Selenium/Remote/Driver.pm

```diff
@@ -236,65 +240,170 @@ more documentation, see the related test methods in L<Selenium::Remote::Driver>
     click_ok
     double_click_ok
 
-=head2 $twd->type_element_ok($search_target, $keys, [, $desc ]);
+=head2 $twd->type_element_ok($search_target [,$locator], $keys, [, $desc ]);
 
-   $twd->type_element_ok( $search_target, $keys [, $desc ] );
+   $twd->type_element_ok( $search_target [,$locator], $keys [, $desc ] );
 
 Use L<Selenium::Remote::Driver/find_element> to resolve the C<$search_target>
-to a web element, and then type C<$keys> into it, providing an optional test
+to a web element and an optional locator, and then type C<$keys> into it, providing an optional test
 label.
 
-Currently, other finders besides the default are not supported for C<type_ok()>.
 
 =cut
 
+# function composing a find_element with locator with a webelement test
+
+sub _find_element_with_action { 
+    my $self = shift; 
+    my $method = shift;
+    my ($locator,$locator_strategy,$params,$desc) = @_;
+    # case 4 args 
+    if ($desc) { 
+        $self->croak('Invalid locator strategy') unless ($self->FINDERS->{$locator_strategy});
+    }
+    else { 
+        if ($params) { 
+            # means that we called it the 'old way' (no locator strategy)
+            if (!$self->FINDERS->{$locator_strategy}) { 
+                $desc = $params; 
+                $params = $locator_strategy; 
+                $locator_strategy = $self->default_finder;
+            }
+        }
+        else { 
+            # means it was called with no locator strategy and no desc 
+            if ($locator_strategy) { 
+                if (!$self->FINDERS->{$locator_strategy}) { 
```

Comment:
this might need to be `exists` instead of `!`? oh, actually, if FINDERS is a constant, maybe it won't be autovivifying $locator_strategy. (double actually, maybe I don't understand autovivification :P)